numba/typeinfer.py

```diff
@@ -1330,8 +1330,8 @@ def typeof_expr(self, inst, target, expr):
                                               src=expr.value.name,
                                               loc=inst.loc))
         elif expr.op == 'make_function':
-            self.lock_type(target.name, types.pyfunc_type, loc=inst.loc,
-                                                            literal_value=expr)
+            self.lock_type(target.name, types.MakeFunctionLiteral(expr),
```

Comment:
It's odd to have the `MakeFunctionLiteral` type references the `ir.Expr` node instead of a literal value for the function.

Reply:
Yes, it seems odd but the `Expr` node has other attributes like `closure` and `defaults` as well. Is this causing an issue potentially? A namedtuple can be kept instead.

Reply:
Being a Literal subclass with full functionality is potentially useful since things like `unliteral` make sense (just pass the function pointer if needed maybe?). Can it cause a problem?

Reply:
I guess we can keep it for now for this usecase to mature a bit.  The impact of this should be small for numba as I do not know of any user code that can trigger it.  We can update this later.

Reply:
I agree.